src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -134,8 +134,78 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return ch
         End Function
 
+        Friend Function TryPeek(ByRef c As Char) As Boolean
+            If _lineBufferOffset >= _bufferLen Then Return False
```

Comment:
@pharring 
Yes it could have been, but I choose not to as this implementation doesn't invoke two additional methods.
eg
```vb
If CanGetCharAt() Then 
   c = Peek()
  Return True
Else
  Return False
End If
```